### PR TITLE
Arachnomania - More spiders

### PR DESCRIPTION
**Pull request by HerCoyote23**
## About the PR

Splits spiders into four different types based on /tg/'s. 
Let me know if the numbers are out of whack.

## Why / Balance

Should make spiders a bit more interesting to interact with. Also builds part of the foundation for a proper /tg/ style spider antag in the future.

Adds two types of 'basic' spiders, and two rarer 'elite' spiders.
(All piercing damage values are adjusted for the direct attack damage bonus. The poisons don't need to be.)

Hunter:
50% of all spider spawns.
70HP
4.3 Movespeed
Deals 6 piercing damage on bite, and injects 0.75u of Toxin, for a sum of 12 damage: 6 piercing and 6 poison.
image

Guard:
45% of all spider spawns.
120HP
4 Movespeed
Deals 8 piercing damage on bite, and injects 0.25u of Toxin, for a sum of 10 damage: 8 piercing and 2 poison.
image

Viper: 
2.5% of all spider spawns.
50HP
4.5 Movespeed
Deals 2.5 piercing damage on bite, and injects 4u of a new reagent - Venom. Venom is a slow-acting poison, 4u is equal to 30 damage over 20 seconds. Best you run off to med after an encounter with this one.
image

Tarantula:
2.5% of all spider spawns.
300HP
2.5 Movespeed
Deals 30 piercing damage on bite, no poisons. Just good old fashioned kill.
Best not to get too close to this one. He's slow and easy to engage in range.
image

Also updates Shiva and Clown spider to inject toxin instead of straight dealing poison damage.
## Technical details

Venom: 
Metabolism Rate: 0.1
Effect: 0.75 poison damage a cycle
1u = 7.5 damage

All the new spiders are now affected by damage slowdown. Let me know if I implemented the thresholds wrong.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Vent spiders have undergone sudden evolution into four distinct archetypes: The more common Hunter and Guard, and the rare and lethal Viper and Tarantula.

### CODE REVIEW

File: Resources/Prototypes/Entities/Mobs/NPCs/spider.yml [hunk 1]
```diff
@@ -0,0 +1,375 @@
+# Code unique spider prototypes or combine them all into one spider and get a
+# random sprite state when you spawn it.
+- type: entity
+  name: base spider
+  parent: SimpleMobBase
+  id: MobBaseSpider
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Mobs/NPCs/spider.yml:6]: add `abstract: true`

**Comment by Jackal298** [comment]
I would like to ask about the queen, but I realized that there will probably be problems with its implementation

File: Resources/Prototypes/Entities/Mobs/NPCs/spider.yml [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+  description: Widely recognized to be the literal worst thing in existence.
+  components:
+  - type: Sprite
+    drawdepth: Mobs
+    layers:
+    - map: ["enum.DamageStateVisualLayers.Base"]
+      state: hunter
+      sprite: Mobs/Animals/spider.rsi
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Mobs/NPCs/spider.yml:14]: you can remove that line for other spiders, and leave only here

File: Resources/Prototypes/Entities/Mobs/NPCs/spider.yml [hunk 3]
```diff
@@ -0,0 +1,375 @@
+# Code unique spider prototypes or combine them all into one spider and get a
+# random sprite state when you spawn it.
+- type: entity
+  name: base spider
+  parent: SimpleMobBase
+  id: MobBaseSpider
+  description: Widely recognized to be the literal worst thing in existence.
+  components:
+  - type: Sprite
+    drawdepth: Mobs
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Mobs/NPCs/spider.yml:10]: it also can be removed from other spiders

File: Resources/Prototypes/Entities/Mobs/NPCs/spider.yml [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+  name: hunter
+  parent: MobHunterSpider
+  id: MobHunterSpiderAngry
+  suffix: Angry
+  components:
+    - type: NpcFactionMember
+      factions:
+        - Xeno
+    - type: InputMover
+    - type: MobMover
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Mobs/NPCs/spider.yml:240]: why not to add it to base spider?
at least `InputMover` and `MobMover`

**Comment by NancokPS2** [comment]
Remila the bat injects poison now?  
Or did you mean Shiva, the sec spider?

**Comment by Jackal298** [comment]
> Remila the bat injects poison now? Or did you mean Shiva, the sec spider?

according to the lines of code, this is shiva

**Comment by Jackal298** [comment]
as sweepers above, correct the error in the description of the PR so as not to confuse

**Comment by Jackal298** [comment]
> as written above, correct the error in the description of the PR so as not to confuse

**Comment by HerCoyote23** [comment]
Done.

**Comment by JoeHammad1844** [comment]
one step closer to the TG spider wars

**Comment by VektorZ1** [comment]
Sssspidersss

**Comment by metalgearsloth** [comment]
```
Tarantula:
5% of all spider spawns.
300HP
2.5 Movespeed
Deals 30 piercing damage on bite, no poisons. Just good old fashioned kill.
Best not to get too close to this one. He's slow and easy to engage in range.
image

Also updates Shiva and Clown spider to inject toxin instead of straight dealing poison damage.
```

Please see recent spider nerfs as this is more than triple what it should be.

**Comment by Jackal298** [comment]
> ```
> Tarantula:
> 5% of all spider spawns.
> 300HP
> 2.5 Movespeed
> Deals 30 piercing damage on bite, no poisons. Just good old fashioned kill.
> Best not to get too close to this one. He's slow and easy to engage in range.
> image
> 
> Also updates Shiva and Clown spider to inject toxin instead of straight dealing poison damage.
> ```
> 
> Please see recent spider nerfs as this is more than triple what it should be.

But after all, an ordinary spider in terms of stats is a tarantula, with this, the tarantula is now rare

**Comment by Jackal298** [comment]
> > ```
> > Tarantula:
> > 5% of all spider spawns.
> > 300HP
> > 2.5 Movespeed
> > Deals 30 piercing damage on bite, no poisons. Just good old fashioned kill.
> > Best not to get too close to this one. He's slow and easy to engage in range.
> > image
> > 
> > Also updates Shiva and Clown spider to inject toxin instead of straight dealing poison damage.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Please see recent spider nerfs as this is more than triple what it should be.
> 
> But after all, an ordinary spider in terms of stats is a tarantula, with this PR, the tarantula is now rare

**Comment by HerCoyote23** [comment]
I did look at the nerfs, Viper and Tarantula are meant to be very rare and very dangerous. Guard and Hunter would take the position that the current tarantula fills.
Might make Viper and Tarantula even more rare though, seeing as they're so dangerous.

**Comment by JoeHammad1844** [comment]
for comparison, what's the default player movespeed? (human with no damage)

**Comment by LankLTE** [comment]
> for comparison, what's the default player movespeed? (human with no damage)

Human runspeed is 4.5 and walkspeed is 2.5. 

So for reference: the hunter moves the same speed as a human running, the guard moves the same as a diona running, the tarantula moves the same as a human walking, and the viper moves the same as arachnids did before their speed was nerfed (.5 faster than humans running).

**Comment by JoeHammad1844** [comment]
so you are basically forced to fight the viper (which is at least offset by their low initial damage, health and spawn rate) and hunter, because doors will slow you down (but if you have anything as robust or more robust than a toolbox you'll out damage and comfortably kill them)

also, facing any two without a firearm is a death sentence

**Comment by JoeHammad1844** [comment]
wait, 3 to 4 bites from a viper is a guaranteed crit without an anti poison medipen

**Comment by Jackal298** [comment]
> wait, 3 to 4 bites from a viper is a guaranteed crit without an anti poison medipen

Yes, this damage is compensated by health so that the antag can at least do something

**Comment by JoeHammad1844** [comment]
who knew balancing is hard

**Comment by HerCoyote23** [comment]
Yeah, 3 bites from Viper would crit you, the catch is that much venom would take a full 60 second to do so unless my math is wrong. In that time you could be getting help or running to med.

Tbh maybe it's not good design to force you to run to med like that. Also new players might not get the memo and just collapse dead for "no reason" after encountering viper.

Maybe I should reduce it's movespeed so you can avoid it? I'm hesitant to change the high damage over a long time cause I think it's thematically/mechanically cool

**Comment by JoeHammad1844** [comment]
maybe make it obvious that you were poisoned, like a green overlay or an ominous message, but forcing you to fight it because of its high movespeed sucks

**Comment by HerCoyote23** [comment]
Just set viper movespeed to 4.5, hopefully that'll make it less problematic. You won't be able to escape super easy, but it should be more doable now.

Slightly increased hunter speed to 4.3

Reduced guard damage from 8 pierce 4 poison to 8 pierce 2 poison, increased their movespeed to 4.
Mostly did that cause being slow REALLY sucks. This way they're slower than humans, but still fast enough to put up a fight.

**Comment by Jackal298** [comment]
I hope it gets merged, your PR gives the spiders development that can grow into their full appearance as a secondary antagonist like in the original SS13

**Comment by HerCoyote23** [comment]
Thanks, that's the hope! Right now I just need to get the numbers in line though. Got any suggestions?

**Comment by Jackal298** [comment]
> Thanks, that's the hope! Right now I just need to get the numbers in line though. Got any suggestions?

It seems to me that now all the characteristics are normal, so that they are dangerous, but not too

**Comment by HerCoyote23** [comment]
Gonna work a bit more on adding broodmother, then PR that.

**Comment by Jackal298** [comment]
I hope that everything works out for you, the spiders are very poorly developed, and I hope that your PR will correct this problem

**Pull request closed without merging**